android/src/main/java/de/atroo/securestorage/EncryptedPreferencesImpl.java

```diff
@@ -0,0 +1,81 @@
+package de.atroo.securestorage;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+import androidx.security.crypto.EncryptedSharedPreferences;
+import androidx.security.crypto.MasterKeys;
+
+import java.util.Map;
+import java.util.Set;
+
+@RequiresApi(api = Build.VERSION_CODES.M)
+public class EncryptedPreferencesImpl implements SecureStorageInterface {
+    private static final String TAG = "EncryptedPrefsImpl";
+    private SharedPreferences sharedPreferences;
+
+    @Override
+    public boolean init(Context context, String preferencesName) {
+        try {
+            String masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC);
+
+            sharedPreferences = EncryptedSharedPreferences.create(
+                    preferencesName,
+                    masterKeyAlias,
+                    context,
+                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+            );
+//            MasterKey masterKey = new MasterKey.Builder(context)
+//                    .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+//                    .build();
+//            Log.d(TAG, "init: " + preferencesName);
+//            sharedPreferences = EncryptedSharedPreferences.create(
+//                    context,
+//                    preferencesName,
+//                    masterKey,
+//                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+//                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+//            );
```

Comment:
@laszloblum WHat about the comment block?

Reply:
Left there unintentionally. Removed